.github/workflows/release.yaml

```diff
@@ -3,59 +3,55 @@ name: Release
 on:
   push:
     tags:
-      - "*"
+      - v[0-9]+.*
 jobs:
-  build:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        rust: [stable]
+  body:
+    runs-on: ubuntu-latest
     permissions:
       contents: write
-
-    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
-        uses: actions/checkout@master
-
-      - name: Install just
-        uses: taiki-e/install-action@just
-
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-
-      - name: Setup build env for Linux
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          sudo apt update
-          sudo apt install -y musl-tools gcc
-          rustup target add x86_64-unknown-linux-musl
-
-      - name: Build for Linux
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          just build-release-linux
-          cp target/x86_64-unknown-linux-musl/release/tenere tenere-x86_64-linux-musl
-
-      - name: Build for macos
-        if: matrix.os == 'macos-latest'
-        run: |
-          rustup target add aarch64-apple-darwin
-          just build-release-macos
-          cp target/x86_64-apple-darwin/release/tenere tenere-x86_64-macos
-          cp target/aarch64-apple-darwin/release/tenere tenere-aarch64-macos
-
+        uses: actions/checkout@v4
       - name: Extract release notes
-        if: matrix.os == 'ubuntu-latest'
+        uses: ffurrer2/extract-release-notes@v2
         id: release_notes
-        uses: ffurrer2/extract-release-notes@v1
+      - name: Upload Body
+        uses: softprops/action-gh-release@v2
+        with:
+          body: ${{ steps.release_notes.outputs.release_notes }}  
 
-      - name: Release
-        uses: softprops/action-gh-release@v1
+  build:
+    permissions:
+      contents: write
+    continue-on-error: true
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+          - target: aarch64-apple-darwin
+            os: macos-latest
+          - target: x86_64-apple-darwin
+            os: macos-latest
+          - target: x86_64-pc-windows-msvc
+            os: windows-latest
+          - target: aarch64-pc-windows-msvc
+            os: windows-latest
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Target
+        run: rustup target add ${{ matrix.target }}
+      - name: Compile
+        run: cargo build --release --target ${{ matrix.target }}
+      - name: Zip Binary (windows)
+        if: runner.os == 'Windows'
+        run: Compress-Archive -Path ./target/${{ matrix.target }}/release/tenere.exe -DestinationPath tenere-${{ matrix.target }}.zip
+      - name: Tar Binary (linux)
+        if: runner.os != 'Windows'
+        run: tar -cJf tenere-${{ matrix.target }}.tar.xz ./target/${{ matrix.target }}/release/tenere
```

Comment:
No need to compress or archive the binaries.

Reply:
There is need. The executable should be called tenere. And we cannot have 5 times just "tenere" in downloads.

"Why not just rename the executable?" You ask. Either way the user needs to rename or unpack the executable at the end. Or worse using scoop having a custom rename script in the json.

Reply:
it is mush simpler just to rename a binary and  I do not see any value of compression.

Reply:
The value is faster downloads.
On GitHub people get rate limited.
Tools like scoop default to think that it's in an archive.

Reply:
Moreover we could actually ship the license and changelog in the archive

Reply:
I personally do not like shipping other stuff other than the binaries.
I still believe that uploading the binaries without any compression or archive is the way to go.